### PR TITLE
Remove all Kyma finalizers when Kyma is under deletion and no secret found

### LINKED ISSUE #816: Remove all Kyma finalizer when Kyma under deleting and no secret found

### PR DESCRIPTION
**Pull request by nesmabadr**
**Description**

Changes proposed in this pull request:

- Remove all finalizers when Kyma is under deletion and no secret found

**Related issue(s)**
Resolves #816

### CODE REVIEW

**Comment by ruanxin** [comment]
I put a hold label just want to have a second review later after you @medmes

File: controllers/kyma_controller.go
Comment (medmes) [inline controllers/kyma_controller.go:426]: If you don't mind, it will be nice to separate the logic between removing `Finalizer` and update the kyma CR. What do you think?

**Review by medmes — APPROVED** [review]

**Review by ruanxin — APPROVED** [review]

**Pull request merged**
